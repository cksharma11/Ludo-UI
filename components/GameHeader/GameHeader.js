import React from 'react';
import PropTypes from 'prop-types';
import GameHeaderStyles from './GameHeader.style';

const GameHeader = ({ playerName }) => {
  return (
    <header>
      <div id="title">Ludo</div>
      <div id="player_name">Hi {playerName}!</div>
      <style jsx>{GameHeaderStyles}</style>
    </header>
  );
};

GameHeader.defaultProps = {
  playerName: ''
};

GameHeader.propTypes = {
  playerName: PropTypes.string
};

export default GameHeader;
