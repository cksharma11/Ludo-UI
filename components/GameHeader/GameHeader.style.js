import css from 'styled-jsx/css';
import colors from '../../config/styles/colors';

const GameHeaderStyles = css`
  header {
    display: flex;
    justify-content: space-between;
    padding: 20px;
    font-size: 20px;
    background: ${colors.palette1.shade1};
    border-bottom: 1px solid white;
  }
`;

export default GameHeaderStyles;
