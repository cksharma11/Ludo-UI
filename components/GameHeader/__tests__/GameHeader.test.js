import React from 'react';
import { shallow } from 'enzyme';
import GameHeader from '../GameHeader';

describe('GameHeader', () => {
  it('should render properly', () => {
    const wrapper = shallow(<GameHeader />);
    expect(wrapper.debug()).toMatchSnapshot();
  });
});
