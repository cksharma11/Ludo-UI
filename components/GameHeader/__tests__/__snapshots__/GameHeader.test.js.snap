// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GameHeader should render properly 1`] = `
"<header className=\\"jsx-186861727\\">
  <div id=\\"title\\" className=\\"jsx-186861727\\" />
  <div id=\\"player_name\\" className=\\"jsx-186861727\\" />
  <JSXStyle id=\\"186861727\\">
    h
    e
    a
    d
    e
    r
    .
    j
    s
    x
    -
    1
    8
    6
    8
    6
    1
    7
    2
    7
    {
    d
    i
    s
    p
    l
    a
    y
    :
    -
    w
    e
    b
    k
    i
    t
    -
    b
    o
    x
    ;
    d
    i
    s
    p
    l
    a
    y
    :
    -
    w
    e
    b
    k
    i
    t
    -
    f
    l
    e
    x
    ;
    d
    i
    s
    p
    l
    a
    y
    :
    -
    m
    s
    -
    f
    l
    e
    x
    b
    o
    x
    ;
    d
    i
    s
    p
    l
    a
    y
    :
    f
    l
    e
    x
    ;
    -
    w
    e
    b
    k
    i
    t
    -
    b
    o
    x
    -
    p
    a
    c
    k
    :
    j
    u
    s
    t
    i
    f
    y
    ;
    -
    w
    e
    b
    k
    i
    t
    -
    j
    u
    s
    t
    i
    f
    y
    -
    c
    o
    n
    t
    e
    n
    t
    :
    s
    p
    a
    c
    e
    -
    b
    e
    t
    w
    e
    e
    n
    ;
    -
    m
    s
    -
    f
    l
    e
    x
    -
    p
    a
    c
    k
    :
    j
    u
    s
    t
    i
    f
    y
    ;
    j
    u
    s
    t
    i
    f
    y
    -
    c
    o
    n
    t
    e
    n
    t
    :
    s
    p
    a
    c
    e
    -
    b
    e
    t
    w
    e
    e
    n
    ;
    p
    a
    d
    d
    i
    n
    g
    :
    2
    0
    p
    x
    ;
    f
    o
    n
    t
    -
    s
    i
    z
    e
    :
    2
    0
    p
    x
    ;
    b
    a
    c
    k
    g
    r
    o
    u
    n
    d
    :
    #
    f
    4
    f
    3
    f
    3
    ;
    b
    o
    r
    d
    e
    r
    -
    b
    o
    t
    t
    o
    m
    :
    1
    p
    x
     
    s
    o
    l
    i
    d
     
    #
    c
    c
    c
    1
    c
    1
    ;
    }
  </JSXStyle>
</header>"
`;
